src/libtls/tls_crypto.c

```diff
@@ -2496,6 +2549,44 @@ int tls_crypto_get_supported_groups(diffie_hellman_group_t **out)
 	return count;
 }
 
+/**
+ * See header.
+ */
+int tls_crypto_get_supported_signatures(tls_version_t version, bool sign,
```

Comment:
Not sure if the `sign` argument is really necessary. The function is only used for tests where we need both verifying and signing, so the PSS_PSS schemes should probably always get filtered. What's missing is a filter via `lib->plugins->has_feature()` (see `get_signature_algorithms()`), i.e. it's currently the potentially supported schemes and not the actually supported ones.